smg-automotive/components-pkg
# fix: unify user

[IN-1317](https://autoricardo.atlassian.net/browse/IN-1317?atlOrigin=eyJpIjoiYmY3OTgxMGYyN2NmNDM5OWI2M2MzZTIxYzcwY2UwYmEiLCJwIjoiaiJ9)

## Motivation and context

Currently, the header takes a specific user object that is not aligned with the user we use everywhere else in the projects. To avoid type errors, it would be good to unify the user between the header and listings/seller

## Before

The header expected user to look like this: 
```
{
  id: number;
  name: string;
  type: UserType;
  accountId: number;
}
```

## After

User is the one from the AuthContext: MergedUser 
```
{
  id: string;
  account_id: string;
  exp: number;
  email: string;
  userType: MappedUserType;
  userName: string;
}
```

## How to test

You can check that the header works as before [here](https://in-1317-listings-web.branch.autoscout24.dev/en) (PR: https://github.com/smg-automotive/listings-web/pull/1450)

Thank you ⚡ 


[IN-1317]: https://autoricardo.atlassian.net/browse/IN-1317?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
